lib/iris/tests/__init__.py

```diff
@@ -183,7 +183,10 @@ def get_data_path(relative_path):
     """
     if not isinstance(relative_path, six.string_types):
         relative_path = os.path.join(*relative_path)
-    data_path = os.path.join(iris.config.TEST_DATA_DIR, relative_path)
+    test_data_dir = iris.config.TEST_DATA_DIR
+    if test_data_dir is None:
```

Comment:
it can occur, here:
https://github.com/SciTools/iris/blob/master/lib/iris/config.py#L138
```
override = os.environ.get("override_test_data_repository")
if override:
    if override == '1':
        TEST_DATA_DIR = None
    else:
        TEST_DATA_DIR = override
```